### PR TITLE
chore: adding API Key guides and suggestions

### PR DESCRIPTION
**Pull request by armsteadj1**
## Description

- Adds more context to the change to Application Keys

## Testing required outside of automated testing?

- [ ] Not Applicable

### Screenshots (if appropriate):

- [ ] Not Applicable

## Rollback / Rollforward Procedure

- [ ] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: docs/Guides/v1_v2_migration.md [hunk 1]
```diff
@@ -20,3 +20,22 @@ production, you should:
 1. Duplicate the `basistheory_reactor` with the same `code` as the corresponding `basistheory_reactor_formula`
 2. Change your production systems to start using the newly created `basistheory_reactor`
 3. Delete the old reactor
+
+## basistheory_application `create_key` and basistheory_application_keys
```
Comment (djejaquino) [inline docs/Guides/v1_v2_migration.md:24]: Where do these show up?

File: docs/Guides/v1_v2_migration.md [hunk 2]
```diff
@@ -20,3 +20,22 @@ production, you should:
...
+
+In `v2` we have introduced the concept of [Application Keys](https://developers.basistheory.com/docs/api/applications/application-keys), which allow multiple keys to be created for an application - enabling 
+you to rotate keys without downtime. With this addition, we've added a `create_key` property to the `basistheory_application` resource 
+which is only used when creating a new application. 
+
+- When creating a new `basistheoyr_application` in v2
```
Comment (djejaquino) [inline docs/Guides/v1_v2_migration.md:30]:
```suggestion
- When creating a new `basistheory_application` in v2
```

File: docs/Guides/v1_v2_migration.md [hunk 3]
```diff
@@ -20,3 +20,22 @@ production, you should:
...
+  - We strongly suggest you leave the `create_key` default
+  - Create a `basistheory_application_key` along with this, to get full lifecycle management of the key. 
+- If you are migrating from a `v1` basistheory_application to `v2`
+  - You will need to update your `basistheory_application` with the `create_key` property set to `true`
```
Comment (djejaquino) [inline docs/Guides/v1_v2_migration.md:34]:
```suggestion
  - You will need to update your existing `basistheory_application` resources with the `create_key` property set to `true`
```

File: docs/Guides/v1_v2_migration.md [hunk 4]
```diff
@@ -20,3 +20,22 @@ production, you should:
...
+  - You are able to also create a `basistheory_application_key` for this application
+  - ⚠️ Keep in mind that these applications will have an Unmanaged Key, which means you will need to manage the key lifecycle yourself in the portal.
```
Comment (djejaquino) [inline docs/Guides/v1_v2_migration.md:36]: Using a bullet point structure to organize this information makes it a bit confusing. Maybe use level 3 headings for the branching "if"?

Screenshot 2024-04-19 at 10 16 01

Comment (djejaquino) [reply docs/Guides/v1_v2_migration.md:36]: Bullet points or numbered lists could be used for steps.

**Review by djejaquino — CHANGES_REQUESTED** [review]

File: docs/Guides/v1_v2_migration.md (hunk 1 above)
Comment (armsteadj1) [reply docs/Guides/v1_v2_migration.md:24]: not sure what you mean?

Comment (djejaquino) [reply docs/Guides/v1_v2_migration.md]: In the repo only. But you are fixing that now. Thanks.

File: docs/guides/v1_v2_migration.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+3. Delete the old reactor
+
+## basistheory_application `create_key` and basistheory_application_keys
+
+In `v2` we have introduced the concept of [Application Keys](https://developers.basistheory.com/docs/api/applications/application-keys), which allow multiple keys to be created for an application - enabling 
+you to rotate keys without downtime. With this addition, we've added a `create_key` property to the `basistheory_application` resource 
+which is only used when creating a new application. 
+
+### Migrating from a `v1` basistheory_application to `v2`
+1. You will need to update your existing `basistheory_application` resource with the `create_key` property set to `true`
```
Comment (djejaquino) [inline docs/guides/v1_v2_migration.md:31]: Does someone have these options, or is it just information?

File: docs/resources/application.md [hunk 6]
```diff
@@ -49,6 +49,7 @@ resource "basistheory_application" "my_application" {
 ### Optional
 
 - `create_key` (Boolean) Create key by default for the Application. Do not set to 'true' if you want to manage the key with the 'basistheory_application_key' resource
+  - We strongly suggest you leave the `create_key` default for new Applications, for existing Applications read our guide on [migrating to v2](/docs/guides/v1_v2_migration.md#basistheory_application-create_key-and-basistheory_application_keys)
```
Comment (djejaquino) [inline docs/resources/application.md:52]:
```suggestion
  - We strongly suggest you to omit `create_key` in new Applications. For existing Applications read our guide on [migrating to v2](/docs/guides/v1_v2_migration.md#basistheory_application-create_key-and-basistheory_application_keys) 
```

**Review by djejaquino — APPROVED** [review]

**Pull request merged**